File: experiments/020-PythonVsJava/cells/java-gradle_1-Empty_free/2022-1c-recuperatorio-parcial1/20260921-204311-76559/output/project/src/main/java/customerimporter/NewZipCode.java
package customerimporter;

import java.util.function.Supplier;

public class NewZipCode extends ZipCode {

    // "B1636BBE": a letter, an old zip code number and three letters
    private static final int SIZE = 8;
    private static final int NUMBER_FIRST_INDEX = 1;
    private static final int NUMBER_LAST_INDEX = 5;
    private static final int ENDING_LETTERS_SIZE = 3;

    private final String zipCode;

    // instance creation

    public static boolean canBeCreatedFrom(String aZipCode) {
        return Character.isLetter(aZipCode.charAt(0));
    }

    // error messages

    public static String invalidZipCodeErrorDescription() {
        return "Invalid new zipcode";
    }

    // initialization

    NewZipCode(String aZipCode) {
        assertIsValid(aZipCode);
        zipCode = aZipCode;
    }

    // zip code

    public Object value() {
        return zipCode;
    }

    // testing

    public boolean isNew() {
        return true;
    }

    public boolean isOld() {
        return false;
    }

    // values

    public Object newZipCodeIfNone(Supplier<Object> aNoneBlock) {
        return zipCode;
    }

    public Object oldZipCodeIfNone(Supplier<Object> aNoneBlock) {
        return aNoneBlock.get();
    }

    // initialization - private

    private static void assertIsValid(String aZipCode) {
        String errorDescription = invalidZipCodeErrorDescription();

        Validation.assertThat(aZipCode.length() == SIZE, errorDescription);
        OldZipCode.assertIsValidNumber(aZipCode.substring(NUMBER_FIRST_INDEX, NUMBER_LAST_INDEX), errorDescription);
        Validation.assertAreAllLetters(aZipCode.substring(aZipCode.length() - ENDING_LETTERS_SIZE), errorDescription);
    }
}

File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2022-1c-recuperatorio-parcial1/20260921-205812-95418/workdir/src/customerimporter/new_zip_code.py
from customerimporter.zip_code import ZipCode


class NewZipCode(ZipCode):

    # error messages

    @classmethod
    def invalid_code_error_description(cls):
        return "Invalid new zipcode"

    # type

    @classmethod
    def is_for(cls, a_zip_code):
        return a_zip_code[0].isalpha()

    # code - validation

    @classmethod
    def assert_is_valid_code(cls, a_zip_code):
        # "B1636BBE" size 8
        cls.assert_valid_code(len(a_zip_code) == 8)
        new_zipcode_old_zip_code = a_zip_code[1:5]
        cls.assert_valid_code(new_zipcode_old_zip_code.isdigit())
        cls.assert_valid_code(int(new_zipcode_old_zip_code) > 999)
        cls.assert_valid_code(a_zip_code[-3:].isalpha())

    # code

    def new_zip_code_if_none(self, a_none_block):
        return self._code

    def old_zip_code_if_none(self, a_none_block):
        return a_none_block()

    def is_new(self):
        return True

    def is_old(self):
        return False
